lib/multi_aggregator/data_transfer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module MultiAggregator
+  class DataTransfer
+    attr_reader(
+      :logger
+    )
+
+    def initialize(
+      logger = MultiAggregator::Logger.new
+    )
+      @logger = logger
+    end
+
+    def call(query_spec, storage, providers = {})
+      query_spec.each do |provider_id, tables_spec|
+        transfer_tables(storage, providers[provider_id], provider_id, tables_spec)
+      end
+    end
+
+    private
+
+    def transfer_tables(storage, provider, db_name, tables_spec)
+      # TODO: move to processor
+      uid = generate_uid
+      storage.uid = uid
+
+      tables_spec.each do |table, columns_spec|
+        transfer_table(storage, provider, db_name, table, columns_spec)
+      end
```

Comment:
Good solution – run it in separated enviroments.

We can add `Thread.new` but we probably will not get desired increasing performance because MRI etc

Also we can use jruby xD